floss/language/go/extract.py

```diff
@@ -0,0 +1,511 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import re
+import sys
+import struct
+import logging
+import pathlib
+import argparse
+from typing import List, Iterable, Optional
+from itertools import chain
+
+import pefile
+
+from floss.results import StaticString, StringEncoding
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 6
+
+
+def extract_strings_from_import_data(pe: pefile.PE) -> List[StaticString]:
+    if not hasattr(pe, "DIRECTORY_ENTRY_IMPORT"):
+        # This is a Go binary, but it doesn't have an import table
+        return []
+
+    import_strings = list()
+
+    for entry in pe.DIRECTORY_ENTRY_IMPORT:
+        for imp in entry.imports:
+            if imp.name is not None:
+                addr = imp.name_offset
+                try:
+                    import_string = StaticString(
+                        string=imp.name.decode("ascii"), offset=addr, encoding=StringEncoding.ASCII
+                    )
+                    import_strings.append(import_string)
+                except UnicodeDecodeError:
+                    continue
+
+    return import_strings
```

Comment:
this is a fine candidate for a generator because it enumerates items and provides increment results.